core/src/main/scala/kafka/log/Log.scala

```diff
@@ -217,6 +218,9 @@ class Log(@volatile var dir: File,
         if (indexFileExists) {
           try {
             segment.index.sanityCheck()
+            // Resize the time index file to 0 if it is newly created.
+            if (!timeIndexFileExists)
+              segment.timeIndex.resize(0)
```

Comment:
I agree with @lindong28. It seems that for the active segment, we should leave the timeindex at the preallocate size.